### PR TITLE
AS-273 Add a passthrough for listSnapshots endpoint in Rawls

### PR DESCRIPTION
**Pull request by zloery**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala]: I don't think the `limit` and `offset` query params are getting passed through to rawls here

Comment (zloery) [reply src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala]: whoops, good catch!

**Review by MatthewBemis — APPROVED** [review]
lgtm

File: src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala:28]: for passthrough apis with a querystring, we typically handle it by passing the querystring as-is. See https://github.com/broadinstitute/firecloud-orchestration/blob/develop/src/main/scala/org/broadinstitute/dsde/firecloud/webservice/WorkspaceApiService.scala#L61 for an example.

Your solution would work too; this is more about consistency in the codebase. If there is a reason to extract/verify only specific params, I'd use your solution; passing it through untouched means we can add/change params in Rawls without needing changes here in orch.

File: src/main/resources/swagger/api-docs.yaml
Comment (davidangb) [inline src/main/resources/swagger/api-docs.yaml]: for namespace and name, use:
```
      parameters:
        - $ref: '#/parameters/workspaceNamespaceParam'
        - $ref: '#/parameters/workspaceNameParam'
```
to normalize the params and cut down on lines of code. See the create-snapshot definition just previous to this as an example.

File: src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala
Comment (zloery) [reply src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala:28]: SGTM - it's better to pass it through untouched in this case

Comment (davidangb) [reply src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala:28]: this should not pass through to `FireCloudConfig.Rawls.baseUrl` - it should pass through to `listDataRepoSnapshotURL(namespace, name)`, and `listDataRepoSnapshotURL` should be ignorant of any querystring.

**Review by davidangb — COMMENTED** [review]
one change needed inline.

Also - do not merge, right? If we merge this it'll make it to prod, and we don't yet have an enable/disable flag for these apis in orchestration.

**Comment by zloery** [comment]
Now that you mention it, it might be important to actually call the right Rawls endpoint...
Also, this is merging to a feature branch so it shouldn't get to production until we merge the [whole branch](https://github.com/broadinstitute/firecloud-orchestration/pull/783), right?

File: src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala]: sorry, one more tweak - need to remove the `pathPrefix("workspaces" / Segment / Segment / "snapshots")` here. This is already inside a `pathPrefix("workspaces" / Segment / Segment / "snapshots")` at line 20. All the nested path directives (including the `pathEnd` at line 21) combine to define the path - this extra line makes it invalid!

Comment (zloery) [reply src/main/scala/org/broadinstitute/dsde/firecloud/webservice/SnapshotApiService.scala]: good catch!

**Review by davidangb — APPROVED** [review]
:+1: ... and you're totally right, this is merging to a feature branch, I retract my previous fear about merging this!

**Pull request merged**
